Integration test failing

Running `examples/deploy-all.sh` currently throws this on my machine.

Not sure what is going on there, it throws it when esbuild tries to list the packages (`npm ls -json -prod -long -all`) in its staging directory `examples/serverless-esbuild/.esbuild/.build` 

```
npm ERR! code ELSPROBLEMS
npm ERR! invalid: esbuild@0.15.1 /Users/seandawson/Development/work/aws-durable-lambda/plugin/node_modules/esbuild
{
  "error": {
    "code": "ELSPROBLEMS",
    "summary": "invalid: esbuild@0.15.1 /Users/seandawson/Development/work/aws-durable-lambda/plugin/node_modules/esbuild",
    "detail": ""
  }
}
```

--- a/examples/assert.sh
+++ b/examples/assert.sh
@@ -1,0 +1,248 @@
+#!/usr/bin/env bash
+
+#####################################################################
+##
+## title: Assert Extension
+##
+## description:
+## Assert extension of shell (bash, ...)
+##   with the common assert functions
+## Function list based on:
+##   http://junit.sourceforge.net/javadoc/org/junit/Assert.html
+## Log methods : inspired by
+##	- https://natelandau.com/bash-scripting-utilities/
+## author: Mark Torok
+##
+## date: 07. Dec. 2016
+##
+## license: MIT
+##
+#####################################################################
+
+if command -v tput &>/dev/null && tty -s; then
+  RED=$(tput setaf 1)
+  GREEN=$(tput setaf 2)
+  MAGENTA=$(tput setaf 5)
+  NORMAL=$(tput sgr0)
+  BOLD=$(tput bold)
+else
+  RED=$(echo -en "\e[31m")
+  GREEN=$(echo -en "\e[32m")
+  MAGENTA=$(echo -en "\e[35m")
+  NORMAL=$(echo -en "\e[00m")
+  BOLD=$(echo -en "\e[01m")
+fi
+
+log_header() {
+  printf "\n${BOLD}${MAGENTA}==========  %s  ==========${NORMAL}\n" "$@" >&2
+}
+
+log_success() {
+  printf "${GREEN}✔ %s${NORMAL}\n" "$@" >&2
+}
+
+log_failure() {
+  printf "${RED}✖ %s${NORMAL}\n" "$@" >&2
+}
+
+
+assert_eq() {
+  local expected="$1"
+  local actual="$2"
+  local msg="${3-}"
+
+  if [ "$expected" == "$actual" ]; then
+    return 0
+  else
+    [ "${#msg}" -gt 0 ] && log_failure "$expected == $actual :: $msg" || true
+    return 1
+  fi
+}
+
+assert_not_eq() {
+  local expected="$1"
+  local actual="$2"
+  local msg="${3-}"
+
+  if [ ! "$expected" == "$actual" ]; then
+    return 0
+  else
+    [ "${#msg}" -gt 0 ] && log_failure "$expected != $actual :: $msg" || true
+    return 1
+  fi
+}
+
+assert_true() {
+  local actual="$1"
+  local msg="${2-}"
+
+  assert_eq true "$actual" "$msg"
+  return "$?"
+}
+
+assert_false() {
+  local actual="$1"
+  local msg="${2-}"
+
+  assert_eq false "$actual" "$msg"
+  return "$?"
+}
+
+assert_array_eq() {
+
+  declare -a expected=("${!1-}")
+  # echo "AAE ${expected[@]}"
+
+  declare -a actual=("${!2}")
+  # echo "AAE ${actual[@]}"
+
+  local msg="${3-}"
+
+  local return_code=0
+  if [ ! "${#expected[@]}" == "${#actual[@]}" ]; then
+    return_code=1
+  fi
+
+  local i
+  for (( i=1; i < ${#expected[@]} + 1; i+=1 )); do
+    if [ ! "${expected[$i-1]}" == "${actual[$i-1]}" ]; then
+      return_code=1
+      break
+    fi
+  done
+
+  if [ "$return_code" == 1 ]; then
+    [ "${#msg}" -gt 0 ] && log_failure "(${expected[*]}) != (${actual[*]}) :: $msg" || true
+  fi
+
+  return "$return_code"
+}
+
+assert_array_not_eq() {
+
+  declare -a expected=("${!1-}")
+  declare -a actual=("${!2}")
+
+  local msg="${3-}"
+
+  local return_code=1
+  if [ ! "${#expected[@]}" == "${#actual[@]}" ]; then
+    return_code=0
+  fi
+
+  local i
+  for (( i=1; i < ${#expected[@]} + 1; i+=1 )); do
+    if [ ! "${expected[$i-1]}" == "${actual[$i-1]}" ]; then
+      return_code=0
+      break
+    fi
+  done
+
+  if [ "$return_code" == 1 ]; then
+    [ "${#msg}" -gt 0 ] && log_failure "(${expected[*]}) == (${actual[*]}) :: $msg" || true
+  fi
+
+  return "$return_code"
+}
+
+assert_empty() {
+  local actual=$1
+  local msg="${2-}"
+
+  assert_eq "" "$actual" "$msg"
+  return "$?"
+}
+
+assert_not_empty() {
+  local actual=$1
+  local msg="${2-}"
+
+  assert_not_eq "" "$actual" "$msg"
+  return "$?"
+}
+
+assert_contain() {
+  local haystack="$1"
+  local needle="${2-}"
+  local msg="${3-}"
+
+  if [ -z "${needle:+x}" ]; then
+    return 0;
+  fi
+
+  if [ -z "${haystack##*$needle*}" ]; then
+    return 0
+  else
+    [ "${#msg}" -gt 0 ] && log_failure "$haystack doesn't contain $needle :: $msg" || true
+    return 1
+  fi
+}
+
+assert_not_contain() {
+  local haystack="$1"
+  local needle="${2-}"
+  local msg="${3-}"
+
+  if [ -z "${needle:+x}" ]; then
+    return 0;
+  fi
+
+  if [ "${haystack##*$needle*}" ]; then
+    return 0
+  else
+    [ "${#msg}" -gt 0 ] && log_failure "$haystack contains $needle :: $msg" || true
+    return 1
+  fi
+}
+
+assert_gt() {
+  local first="$1"
+  local second="$2"
+  local msg="${3-}"
+
+  if [[ "$first" -gt  "$second" ]]; then
+    return 0
+  else
+    [ "${#msg}" -gt 0 ] && log_failure "$first > $second :: $msg" || true
+    return 1
+  fi
+}
+
+assert_ge() {
+  local first="$1"
+  local second="$2"
+  local msg="${3-}"
+
+  if [[ "$first" -ge  "$second" ]]; then
+    return 0
+  else
+    [ "${#msg}" -gt 0 ] && log_failure "$first >= $second :: $msg" || true
+    return 1
+  fi
+}
+
+assert_lt() {
+  local first="$1"
+  local second="$2"
+  local msg="${3-}"
+
+  if [[ "$first" -lt  "$second" ]]; then
+    return 0
+  else
+    [ "${#msg}" -gt 0 ] && log_failure "$first < $second :: $msg" || true
+    return 1
+  fi
+}
+
+assert_le() {
+  local first="$1"
+  local second="$2"
+  local msg="${3-}"
+
+  if [[ "$first" -le  "$second" ]]; then
+    return 0
+  else
+    [ "${#msg}" -gt 0 ] && log_failure "$first <= $second :: $msg" || true
+    return 1
+  fi
+}

--- a/examples/serverless-esbuild/package.json
+++ b/examples/serverless-esbuild/package.json
@@ -8,9 +8,7 @@
     "node": ">=14.15.0"
   },
   "dependencies": {
-    "@agiledigital/aws-durable-lambda": "../../plugin",
-    "aws-sdk": "^2.1042.0",
-    "dynamoose": "^2.8.3"
+    "@agiledigital/aws-durable-lambda": "../../plugin"
   },
   "devDependencies": {
     "@serverless/typescript": "3.3.0",

--- a/examples/serverless-esbuild/serverless.ts
+++ b/examples/serverless-esbuild/serverless.ts
@@ -31,7 +31,7 @@ const serverlessConfiguration: AWS = {
   },
   custom: {
     esbuild: {
-      external: ['@agiledigital/aws-durable-lambda'],
+      packager: 'yarn',
       bundle: true,
       minify: false,
       sourcemap: true,

--- a/examples/serverless-webpack/serverless.ts
+++ b/examples/serverless-webpack/serverless.ts
@@ -32,6 +32,7 @@ const serverlessConfiguration: AWS = {
     webpack: {
       // TODO: Work out why includeModules/node externals doesn't work properly with ADL plugin
       packager: 'yarn',
+      includeModules: true,
     },
   },
 };

--- a/examples/serverless-webpack/serverless.ts
+++ b/examples/serverless-webpack/serverless.ts
@@ -30,7 +30,6 @@ const serverlessConfiguration: AWS = {
   },
   custom: {
     webpack: {
-      // TODO: Work out why includeModules/node externals doesn't work properly with ADL plugin
       packager: 'yarn',
       includeModules: true,
     },

--- a/examples/serverless-webpack/webpack.config.js
+++ b/examples/serverless-webpack/webpack.config.js
@@ -2,39 +2,41 @@
 const path = require('path');
 const slsw = require('serverless-webpack');
 const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
+const nodeExternals = require('webpack-node-externals');
 
 const isLocal = slsw.lib.webpack.isLocal;
 
 module.exports = {
-    mode: isLocal ? 'development' : 'production',
-    entry: slsw.lib.entries,
-    devtool: 'source-map',
-    resolve: {
-        extensions: ['.js', '.jsx', '.json', '.ts', '.tsx']
-    },
-    output: {
-        libraryTarget: 'commonjs2',
-        path: path.join(__dirname, '.webpack'),
-        filename: '[name].js'
-    },
-    target: 'node',
-    module: {
-        rules: [
-            {
-                // Include ts, tsx, js, and jsx files.
-                test: /\.(ts|js)x?$/,
-                exclude: /node_modules/,
-                use: [
-                    {
-                        loader: 'cache-loader',
-                        options: {
-                            cacheDirectory: path.resolve('.webpackCache')
-                        }
-                    },
-                    'babel-loader'
-                ]
-            }
-        ]
-    },
-    plugins: [new ForkTsCheckerWebpackPlugin()]
+  mode: isLocal ? 'development' : 'production',
+  entry: slsw.lib.entries,
+  devtool: 'source-map',
+  externals: [nodeExternals()],
+  resolve: {
+    extensions: ['.js', '.jsx', '.json', '.ts', '.tsx'],
+  },
+  output: {
+    libraryTarget: 'commonjs2',
+    path: path.join(__dirname, '.webpack'),
+    filename: '[name].js',
+  },
+  target: 'node',
+  module: {
+    rules: [
+      {
+        // Include ts, tsx, js, and jsx files.
+        test: /\.(ts|js)x?$/,
+        exclude: /node_modules/,
+        use: [
+          {
+            loader: 'cache-loader',
+            options: {
+              cacheDirectory: path.resolve('.webpackCache'),
+            },
+          },
+          'babel-loader',
+        ],
+      },
+    ],
+  },
+  plugins: [new ForkTsCheckerWebpackPlugin()],
 };

--- a/plugin/esbuild.config.js
+++ b/plugin/esbuild.config.js
@@ -1,15 +1,14 @@
-const esbuild = require('esbuild')
+const pkg = require('./package.json');
+const esbuild = require('esbuild');
 
-// Automatically exclude all node_modules from the bundled version
-const { nodeExternalsPlugin } = require('esbuild-node-externals')
-
-esbuild.build({
+esbuild
+  .build({
     entryPoints: [
-        './src/functions/getTask/handler.ts',
-        './src/functions/orchestrator/handler.ts',
-        './src/functions/reporter/handler.ts',
-        './src/functions/submitTask/handler.ts',
-        './src/aws-durable-lambda.js'
+      './src/functions/getTask/handler.ts',
+      './src/functions/orchestrator/handler.ts',
+      './src/functions/reporter/handler.ts',
+      './src/functions/submitTask/handler.ts',
+      './src/aws-durable-lambda.js',
     ],
     outdir: 'lib/',
     bundle: true,
@@ -17,5 +16,6 @@ esbuild.build({
     platform: 'node',
     sourcemap: true,
     target: 'node14',
-    plugins: [nodeExternalsPlugin()]
-}).catch(() => process.exit(1))
+    external: ['aws-sdk'],
+  })
+  .catch(() => process.exit(1));

--- a/plugin/package.json
+++ b/plugin/package.json
@@ -10,8 +10,8 @@
   "scripts": {
     "prepare": "run-s build",
     "build": "rimraf ./lib && tsc && node ./esbuild.config.js",
-    "format": "prettier --write src",
-    "format-check": "prettier --check src"
+    "format": "prettier --write src esbuild.config.js package.json tsconfig.json tsconfig.paths.json",
+    "format-check": "prettier --check src esbuild.config.js package.json tsconfig.json tsconfig.paths.json"
   },
   "engines": {
     "node": ">=14.15.0"

--- a/plugin/package.json
+++ b/plugin/package.json
@@ -18,11 +18,8 @@
   },
   "dependencies": {
     "fs-extra": "^10.1.0",
-    "uuid": "^8.3.2"
-  },
-  "peerDependencies": {
-    "aws-sdk": ">=2.1042.0",
-    "dynamoose": ">=2.8.3"
+    "uuid": "^8.3.2",
+    "dynamoose": "3.0.0"
   },
   "devDependencies": {
     "@serverless/typescript": "3.21.0",
@@ -31,7 +28,6 @@
     "@types/node": "16.11.52",
     "@types/uuid": "8.3.4",
     "aws-sdk": "2.1199.0",
-    "dynamoose": "3.0.0",
     "esbuild": "0.15.5",
     "esbuild-node-externals": "1.4.1",
     "esbuild-node-tsc": "2.0.2",

